src/main/java/net/starschema/clouddb/jdbc/BQConnection.java

```diff
@@ -197,7 +196,12 @@ public BQConnection(String url, Properties loginProp) throws SQLException {
         //do we have a serviceaccount to connect with?
         if (serviceAccount) {
             try {
-                this.bigquery = Oauth2Bigquery.authorizeviaservice(userId, userKey);
+                // Support for old behavior, passing no actual password, but passing the path as 'password'
+                if (userPath == null) {
+                    userPath = userKey;
+                    userKey = null;
+                }
+                this.bigquery = Oauth2Bigquery.authorizeviaservice(userId, userPath, userKey);
```

Comment:
Yeah, I can't say I LIKE this, but it does preserve backwards compatibility. 

We probably want to produce a README soon. As things are starting to change and the docs are non-existent.